common/src/main/java/com/wynntils/core/net/athena/WynntilsAccountManager.java

```diff
@@ -86,55 +84,45 @@ public void dumpEncodedConfig(String name) {
     }
 
     private static void doLogin() {
-        if (WebManager.getApiUrls().isEmpty() || !WebManager.getApiUrls().get().hasKey("Athena")) return;
-
-        WebReader webReader = WebManager.getApiUrls().get();
-        RequestHandler handler = WebManager.getHandler();
-
-        String baseUrl = webReader.get("Athena");
-        String[] secretKey = new String[1]; // it's an array for the lambda below be able to set its value
-
+        CompletableFuture<Boolean> future = new CompletableFuture<>();
         // generating secret key
+        ApiResponse apiResponse = NetManager.callApi(UrlId.API_ATHENA_AUTH_PUBLIC_KEY);
+        apiResponse.handleJsonObject(json -> {
+            String secretKey = parseAndJoinPublicKey(json.get("publicKeyIn").getAsString());
+
+            Map<String, String> arguments = new HashMap<>();
+            arguments.put("key", secretKey);
+            arguments.put("username", McUtils.mc().getUser().getName());
+            arguments.put("version", String.format("A%s %s", WynntilsMod.getVersion(), WynntilsMod.getModLoader()));
+
+            ApiResponse apiAuthResponse = NetManager.callApi(UrlId.API_ATHENA_AUTH_RESPONSE, arguments);
+            apiAuthResponse.handleJsonObject(authJson -> {
+                token = authJson.get("authToken").getAsString(); /* md5 hashes*/
+                JsonObject hashes = authJson.getAsJsonObject("hashes");
+                hashes.entrySet()
+                        .forEach((k) ->
+                                md5Verifications.put(k.getKey(), k.getValue().getAsString())); /* configurations*/
+                JsonObject configFiles = authJson.getAsJsonObject("configFiles");
+                configFiles
+                        .entrySet()
+                        .forEach((k) ->
+                                encodedConfigs.put(k.getKey(), k.getValue().getAsString()));
+                loggedIn = true;
+                future.complete(true);
+                WynntilsMod.info("Successfully connected to Athena!");
+            });
+        });
+
+        // FIXME: Rewrite athena account manager dependencies to work async
```

Comment:
Didn't the commit description just say this is now up to date? Or what is this about?

Reply:
Is this about the users of the token does not respect loading this is async, hence making the code weird?

Reply:
> Is this about the users of the token does not respect loading this is async, hence making the code weird?

Yes, exactly. Code like Hades is assuming that when the HadesManager or whatever it was called is being init'ed, that the login is done so we have a token. If not, it complains loudly and refuses to make any further attempts at connecting.

I think we should rewrite this so that we send out like a AthenaLoginEvent when we are authenticated with Athena, and that in turn could trigger Hades to log in, etc.

But that was not something I wanted to address here.

Reply:
Agreed. I think we should have an issue for this though.

Reply:
https://github.com/Wynntils/Artemis/issues/829